alibi_detect/saving/schemas.py

```diff
@@ -53,6 +52,20 @@ def coerce_int2list(value: int) -> List[int]:
         return value
 
 
+def validate_model(model: Callable, values: dict) -> Callable:
+    """Validator to check the model is compatible with the given backend"""
+    backend = values['backend']
+    if backend == 'tensorflow':
+        assert isinstance(model, SupportedModels_tf)
+    elif backend == 'pytorch':
+        assert isinstance(model, SupportedModels_torch)
+    elif backend == 'sklearn':
+        assert isinstance(model, SupportedModels_sklearn)
+    return model
+
+
+# TODO - we could add another validator to check given "backend" against what optional deps are installed?
```

Comment:
Not sure what you mean, if the specified backend is missing some dependencies then the user wouldn't be able to instatiate the detector which happens way before saving?

Reply:
Yea the main thing was that I didn't understand the comment :)